### PR TITLE
Deterministic Bundle Trees

### PR DESCRIPTION
**Pull request by brandon93s**
Potential fix for #219.

`Promise.all` guarantees order. This delays the `set` operations until after processing so that they always occur in the same order.

### CODE REVIEW

File: src/Bundler.js [hunk 1]
```diff
@@ -323,22 +323,33 @@ class Bundler extends EventEmitter {
...
-          asset.depAssets.set(dep.name, assetDep);
+          applyFunction = () => {
+            asset.dependencies.set(dep.name, dep);
+            asset.depAssets.set(dep.name, assetDep);
+          };
           await this.loadAsset(assetDep);
         }
+        return applyFunction;
       })
     );
```
Comment (pomle) [inline src/Bundler.js]: Is `applyFunctions` only job to be run later. Then would it be leaner to chain `.forEach(fn => fn())` do you think?

**Comment by pomle** [comment]
👍

File: src/Bundler.js (hunk 1 above)
Comment (brandon93s) [reply src/Bundler.js]: That would work as well. I felt the current was more readable.

Comment (pomle) [reply src/Bundler.js]: All good. Confirmed test is stable on my end now.

**Comment by devongovett** [comment]
Thanks for looking into this! I refactored it slightly to be two loops: one that resolves and loads assets, and the second which assigns them back to deps. https://github.com/parcel-bundler/parcel/commit/539ade1820c3e644586e3161f6b2357a68784142

**Comment by brandon93s** [comment]
Awesome, definitely a bit cleaner. Hopefully the last we see of this test failing!

**Pull request closed without merging**